### PR TITLE
Fix module dependency for intrinsic function return type

### PR DESCRIPTION
**Pull request by Shaikh-Ubaid**
fixes https://github.com/lfortran/lfortran/issues/1673

### CODE REVIEW

**Comment by Shaikh-Ubaid** [comment]
The example shared at https://github.com/lfortran/lfortran/issues/1673#issuecomment-1556040062 now works

```bash
$ cat examples/expr2.f90 
module vtk90m

   private
   public vtk90

   contains

      subroutine vtk90(name, description, label, nx, ny, nz, dx, dy, dz, &
      &  ox, oy, oz, x)

         implicit none

         character(len=*), intent(in) :: name, description, label
         integer, intent(in) :: nx, ny, nz
         real(4), intent(in) :: dx, dy, dz, ox, oy, oz, x(nx,ny,nz)

         integer fd
         integer i, j, k
         character(len=256) desc

         open(newunit=fd, file=trim(name), status='replace')
         write(fd, '(A)') '# vtk DataFile Version 3.0'

         desc = trim(adjustl(description))
         write(fd, '(A)') trim(desc)

         write(fd, '(A)') 'ASCII'
         write(fd, '(A)') 'DATASET STRUCTURED_POINTS'
         write(fd, '(A,1X,I0,1X,I0,1X,I0)') 'DIMENSIONS', nx, ny, nz
         write(fd, '(A,1X,F6.3,1X,F6.3,1X,F6.3)') 'ORIGIN', ox, oy, oz
         write(fd, '(A,1X,ES11.4,1X,ES11.4,1X,ES11.4)') 'SPACING', dx, dy, dz
         write(fd, '(A,1X,I0)') 'POINT_DATA', nx * ny * nz
         write(fd, '(A)') 'SCALARS ' // trim(label) // ' float'
         write(fd, '(A)') 'LOOKUP_TABLE default'
         do k = 1, nz
            do j = 1, ny
               do i = 1, nx
                  write(fd,'(ES12.5)') x(i,j,k)
               end do
            end do
         end do
         close(fd)
      end subroutine vtk90
end module vtk90m
$ lfortran -c examples/expr2.f90
$ 
```

File: tests/reference/asr-intrinsics_36-1cb3d03.stdout [hunk 1]
```diff
@@ -330,7 +330,8 @@
                                 )
                         })
                     intrinsics_36
-                    [lfortran_intrinsic_string]
+                    [lfortran_intrinsic_string
+                    lfortran_intrinsic_builtin]
```
Comment (Shaikh-Ubaid) [inline tests/reference/asr-intrinsics_36-1cb3d03.stdout:334]: It seems that we do not yet robustly check the program dependencies in `asr_verfiy()`. Do we need to add checks similar to that of `Modules` for the dependencies in the `Program`?

Comment (Shaikh-Ubaid) [reply tests/reference/asr-intrinsics_36-1cb3d03.stdout:334]: I just tested adding dependency checks similar to that of `visit_Module` in `visit_Program` of `asr_verfiy()` and many test cases seem to fail.

**Comment by Shaikh-Ubaid** [comment]
Ready.

File: tests/reference/asr-intrinsics_36-1cb3d03.stdout (hunk 1 above)
Comment (certik) [reply tests/reference/asr-intrinsics_36-1cb3d03.stdout:334]: Yes, we should make our verify() much more strict. It will reveal and catch tons of bugs.

**Review by certik — APPROVED** [review]
I think this looks good.

**Pull request merged**